### ISSUE TITLE
Downloads API (v1) returns release files for "hidden" releases

### ISSUE THREAD

**Issue by jaap3** [python/pythondotorg#1308]
While trying to recreate a realistic downloads section for local pythondotorg development I noticed that the following. 

The API v1 downloads [release_file endpoint](https://www.python.org/api/v1/downloads/release_file/?format=json) lists a number of files that refer to a release that seems to be hidden:

* Release file ids [2012](https://www.python.org/api/v1/downloads/release_file/2012/?format=json)-[2022](https://www.python.org/api/v1/downloads/release_file/2022/?format=json) refer to release id [265](https://www.python.org/api/v1/downloads/release/265/?format=json)
* Release file ids [2058](https://www.python.org/api/v1/downloads/release_file/2058/?format=json)-[2062](https://www.python.org/api/v1/downloads/release_file/2062/?format=json)) refer to release id [268](https://www.python.org/api/v1/downloads/release/268/?format=json)  

Accessing these release endpoins results in a 401 Unauthorized response.

I'm not sure what's the cause of this, but it seems that these files should not be listed (publicly) if the release is made unavailable.

**Comment by jaap3** [comment]
On a slightly related note, it seems that API v2 is not available (publicly) yet? e.g. accessing v2 urls results in a 404.

**Comment by berkerpeksag** [comment]
Correct, this is a known issue. I didn't want to change the behavior when I wrote API v2:

https://github.com/python/pythondotorg/blob/4135738b0a1473b3f7bf33099737ace5e9893020/downloads/tests/test_views.py#L424-L432

> On a slightly related note, it seems that API v2 is not available (publicly) yet?

Correct, it isn't deployed yet.
